sw/host/opentitantool/src/command/spi.rs

```diff
@@ -305,6 +302,86 @@ impl CommandDispatch for SpiTpm {
     }
 }
 
+/// Read plain data bytes from a SPI device (not necessarily SPI EEPROM/flash).
+#[derive(Debug, StructOpt)]
+pub struct SpiRawRead {
+    #[structopt(short = "n", long, help = "Number of bytes to read.")]
+    length: usize,
+}
+
+#[derive(Debug, serde::Serialize)]
+pub struct SpiRawReadResponse {
+    hexdata: String,
+}
+
+impl CommandDispatch for SpiRawRead {
+    fn run(
+        &self,
+        context: &dyn Any,
+        transport: &TransportWrapper,
+    ) -> Result<Option<Box<dyn Annotate>>> {
+        transport.capabilities()?.request(Capability::SPI).ok()?;
+        let context = context.downcast_ref::<SpiCommand>().unwrap();
+        let spi_bus = context.params.create(transport, "BOOTSTRAP")?;
+        let mut v = vec![0u8; self.length];
+        spi_bus.run_transaction(&mut [Transfer::Read(&mut v)])?;
+        Ok(Some(Box::new(SpiRawReadResponse {
+            hexdata: hex::encode(v),
+        })))
+    }
+}
+
+/// Write plain data bytes to a SPI device (not necessarily SPI EEPROM/flash).
+#[derive(Debug, StructOpt)]
+pub struct SpiRawWrite {
+    #[structopt(short, long, help = "Hex data bytes to write.")]
+    hexdata: String,
+}
+
+impl CommandDispatch for SpiRawWrite {
+    fn run(
+        &self,
+        context: &dyn Any,
+        transport: &TransportWrapper,
+    ) -> Result<Option<Box<dyn Annotate>>> {
+        transport.capabilities()?.request(Capability::SPI).ok()?;
+        let context = context.downcast_ref::<SpiCommand>().unwrap();
+        let spi_bus = context.params.create(transport, "BOOTSTRAP")?;
+        spi_bus.run_transaction(&mut [Transfer::Write(&hex::decode(&self.hexdata)?)])?;
+        Ok(None)
+    }
+}
+
+/// Write plain data bytes to a SPI device (not necessarily SPI EEPROM/flash).
+#[derive(Debug, StructOpt)]
+pub struct SpiRawWriteRead {
```

Comment:
Is this a Write followed by a Read, or a a simultaneous "Transceive" operation?

Reply:
The though occurred to me, but then I forgot.  This is write followed by read, maybe it should be called `opentitantool spi raw-write-then-read`.  I can also add a `spi raw-transcieve` for completeness.

Reply:
I don't have a problem with either `raw-write-read` or `raw-write-then-read`; I prefer the former for being shorter.  Just needed some clarification on exactly what it was meant to do.

Reply:
Ok, let's go with the shorter names.